packages/form-core/src/form-group/FormGroupMixin.js

```diff
@@ -425,20 +428,50 @@ const FormGroupMixinImplementation = superclass =>
     }
 
     /**
-     * @param {FormControl} child
+     * Traverses the _parentFormGroup tree, and gathers all aria description elements
+     * (feedback and helptext) that should be provided to children.
+     *
+     * In the example below, when the input for 'street' has focus, a screenreader user
+     * would hear the #group-error.
+     * In case one of the inputs was in error state as well, the SR user would
+     * first hear the local error, followed by #group-error
+     * @example
+     * <lion-fieldset name="address">
+     *   <lion-input name="street" label="Street" .modelValue="${'Park Avenue'}"></lion-input>
+     *   <lion-input name="number" label="Number" .modelValue="${100}">...</lion-input>
+     *   <div slot="feedback" id="group-error">
+     *      Park Avenue only has numbers up to 80
+     *   </div>
+     * </lion-fieldset>
      */
-    __linkChildrenMessagesToParent(child) {
-      // aria-describedby of (nested) children
+    __getAllDescriptionElementsInParentChain() {
       const unTypedThis = /** @type {unknown} */ (this);
       let parent = /** @type {FormControlHost & { _parentFormGroup:any }} */ (unTypedThis);
-      const ctor = /** @type {typeof FormGroupMixin} */ (this.constructor);
       while (parent) {
-        ctor._addDescriptionElementIdsToField(child, parent._getAriaDescriptionElements());
+        const descriptionElements = parent._getAriaDescriptionElements();
+        const orderedEls = getAriaElementsInRightDomOrder(descriptionElements, { reverse: true });
+        orderedEls.forEach(el => {
+          this.__descriptionElementsInParentChain.add(el);
+        });
         // Also check if the newly added child needs to refer grandparents
         parent = parent._parentFormGroup;
       }
     }
 
+    /**
+     * @param {FormControl} child
+     * @param {{ unlink?:boolean }} [options]
+     */
+    __linkOrUnlinkParentMessages(child, { unlink = false } = {}) {
```

Comment:
I think this would be more readable as 2 functions

Reply:
Me as well :) Fair point

Reply:
I think as well :) FIxed 👍